spec/services/form_validator_spec.rb

```diff
@@ -128,5 +128,37 @@
       end
     end
   end
+
+  context 'with zip as data type' do
+    let(:element) { create :form_element, :postal, form: form }
+    let(:country_element) { create :form_element, :country, form: form}
+    let(:us_zip) { '12345' }
+    let(:uk_zip) { 'CR0 3RL' }
+    let(:params) { { form_id: element.form_id, postal: us_zip } }
+
+    context 'is valid' do
+      it 'without a country code' do
+        expect(subject).to be_valid
+      end
+
+      it 'with a valid country code in the UK' do
+        params.merge!(postal: uk_zip, country: :UK)
+        expect(subject).to be_valid
+      end
+    end
+
+    context 'is invalid' do
+      it 'with an incorrect code' do
+        params.merge!(postal: 'Not a valid zip')
+        expect(subject).to_not be_valid
+      end
+
+      it 'with a valid code but incorrect country code' do
+        country_element
+        params.merge!(country: :UK)
+        expect(subject).to_not be_valid
+      end
+    end
+  end
```

Comment:
you have logic you added to the `FormValidator` class specifically to make it handle the `nil` case. Right now, you could rip that logic out, so that it would in practice fail when calling your method, but all the specs would still pass. the idea is to cover that edge case with a spec.